modules/stochastic_tools/test/tests/surrogates/load_store/tests

```diff
@@ -10,15 +10,57 @@
       check_files = 'train_out_poly_chaos.rd'
       allow_test_objects = true
 
-      detail = "trained and"
+      detail = "trained with replicated stochastic data and"
     []
     [evaluate]
       type = CSVDiff
       input = evaluate.i
       csvdiff = evaluate_out_pc_moments_0002.csv
       prereq = surrogate/train # Creates data files that this test uses
       allow_warnings = true
-      detail = "evaluated separately."
+      detail = "evaluated separately;"
     []
+
+    [train_distributed]
+      type = CheckFiles
+      input = train.i
+      cli_args = VectorPostprocessors/storage/parallel_type=DISTRIBUTED
+      check_files = 'train_out_poly_chaos.rd'
+      prereq = surrogate/evaluate
+      allow_test_objects = true
+
+      detail = "trained with distributed stochastic data and"
+    []
+    [evaluate_distributed]
+      type = CSVDiff
+      input = evaluate.i
+      csvdiff = evaluate_out_pc_moments_0002.csv
+      prereq = surrogate/train_distributed # Creates data files that this test uses
+      allow_warnings = true
+      detail = "evaluated separately;"
+    []
+
+    [train_three_ranks]
+      type = CheckFiles
+      input = train.i
+      cli_args = VectorPostprocessors/storage/parallel_type=DISTRIBUTED
+      check_files = 'train_out_poly_chaos.rd'
+      prereq = surrogate/evaluate_distributed
+      allow_test_objects = true
+      min_parallel = 3
+
+      detail = "trained with distributed stochastic data and"
+    []
+    [evaluate_two_ranks]
+      type = CSVDiff
+      input = evaluate.i
+      csvdiff = evaluate_out_pc_moments_0002.csv
+      prereq = surrogate/train_three_ranks # Creates data files that this test uses
+      allow_warnings = true
+      max_parallel = 2
+
+      detail = "evaluated separately with a different number of processors."
```

Comment:
So it works with different number of processors?

Reply:
Yes, and with replicated or distributed data.